exercises/practice/promises/.docs/introduction.md

```diff
@@ -0,0 +1,25 @@
+# Introduction
+
+Before the `Promise` class was introduced, there was only one way to deal with asynchronous code : the _callback pattern_.
+
+A callback is a function that accept an "error" as first parameter (see example below)
+
+```javascript
+function callback(error, arg2, arg3) {}
+```
+
+Now is it related to asynchronous code ?
+
+Historically, callbacks have been used in order to allow people to do some work after an asynchronous task was done and without blocking the whole program.
+
+```javascript
+fetchProduct(productId, function (error, data) {
+  if (error !== null) {
```

Comment:
I am no expert on this pattern but shouldn't this be `if (error)` so it also works if error is `undefined` for example?

Reply:
Good catch, fixed in 614fa7060fd3cb084c481a90fef2ed0772c798bb